source/Plugins/ObjectFile/ELF/ObjectFileELF.cpp

```diff
@@ -3481,3 +3482,46 @@ ObjectFileELF::GetLoadableData(Target &target) {
   }
   return loadables;
 }
+
+bool ObjectFileELF::SaveCore(const lldb::ProcessSP &process_sp,
+                             const FileSpec &outfile, Status &error) {
+  if (!process_sp)
+    return false;
+
+  Target &target = process_sp->GetTarget();
+  const ArchSpec target_arch = target.GetArchitecture();
+  const llvm::Triple &target_triple = target_arch.GetTriple();
+
+  // To generate a core, just write the name of the core file, follow by the
+  // name of the current executable at the dpu_save_core_addr
+  if (target_triple.getArch() == llvm::Triple::dpu) {
+
+    // Get the current executable module
+    Module *exe_module = target.GetExecutableModulePointer();
+    if (!exe_module)
+      return false;
+
+    // Get the current executable name
+    char exe_path[PATH_MAX];
+    bool exe_valid = false;
+    exe_valid = exe_module->GetFileSpec().GetPath(exe_path, sizeof(exe_path));
+    if (!exe_valid)
+      return false;
+
+    // Assemble the core file name and the current executable name
+    const char *core_file_path = outfile.GetCString();
+    const size_t core_file_path_len = strlen(core_file_path);
+    const size_t exe_path_len = strlen(exe_path);
+    memcpy(&exe_path[exe_path_len + 1], core_file_path, core_file_path_len);
```

Comment:
Looks like plain old C rather than C++
you can do smaller and better.
first you can avoid good old char*
e.g std::string exe_path = exe_module->GetFileSpec().GetPath(); should work ?

I don't fully understand your assembly:
you want exe full path + only filename for core (not the full path ?) ?

At the end of the day, you can likely work directly with the FileSpec 
e.g. 
outfile.PrependPathComponent(exe_module->GetFileSpec().GetDirectory());
or whatever you are trying to achieve